### ISSUE TITLE
RollupDateLiteralTest Failing at Fiscal Quarter

### ISSUE THREAD

**Issue by jamessimone** [jamessimone/apex-rollup#325]
### Discussed in https://github.com/jamessimone/apex-rollup/discussions/324

<div type='discussions-op-text'>

<sup>Originally posted by **shane-saltbox** May 24, 2022</sup>
@jamessimone I'm attaching a quick analysis of what we found while trying to manually implement the test coverage for the package. We'd like to be able to use the plug-in now that I know it's there, but suspect that'll fail until we're able to address the issue we found. 

For context, our business starts it's fiscal year in February, which has also been updated on the 'Fiscal Year Starts In' value to February. 

While going through the install of the test classes several methods in `RollupDateLiteralTest` failed due to the use of `THIS_QUARTER` instead of `THIS_FISCAL_QUARTER` for example. We were able to get around most of them quickly by utilizing the fiscal date, but also had to make a few other changes / comment a few sections out. At the end of the day there was still a good code coverage after these changes. 

I'm attaching a quick diff so it's easy for you to find what changes we had to make. 

![screencapture-github-shane-saltbox-saltbox-sfdc-pull-43-files-2022-05-24-18_34_22](https://user-images.githubusercontent.com/89033396/170149588-62d921e0-c8cf-4ce5-bbb3-e41a23163ac8.png)
</div>

**Comment by jamessimone** [comment]
@ssk42 let me know if you'd like to tackle this one - otherwise I will plan to take a look once I've addressed the remaining issues from #321. Thanks!

**Comment by jamessimone** [comment]
Jotting down some notes as I was able to take a quick look at this on Friday - I doubt the fixes will make it into the next version, but it will be my priority to fix this once the latest version has been released.

**Comment by jamessimone** [comment]
Making progress on this issue. I've found a number of edge cases with the existing `RollupDateLiteral` implementation - primarily dealing with date literals in SOQL being interpreted using the user's local time - that were exposed while re-working the quarter/fiscal quarter sections, and have led to a number of cascading changes affecting a large number of the other date literals.

TL;DR - progress is being made, but slower than I would like and, if there continue to be other fixes being pinned to `v1.5.9`, I will once again bump this fix to the next version.
